src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -191,6 +191,27 @@ private String escapeSqlQuotes(String originalString) {
     return originalString.replace("\"", "\"\"");
   }
 
+  /**
+   * This guards against SQL injections by ensuring that any single quote is escaped properly.
+   *
+   * @param arg the original schema
+   * @return
+   */
+  private String escapeSingleQuoteForLikeCommand(String arg) {
+    int i = 0;
+    int index = arg.indexOf("'", i);
+    while (index != -1) {
+      if (index == 0 || (index > 0 && arg.charAt(index - 1) != '\\')) {
```

Comment:
we shouldn't replace first quote  i.e. very begining of the data, as escape one. e.g. arg value can be 'abcd', we should keep as is. is there is a quote inside then escape it. e.g. 'abcd'xyz' can be change as 'abcd\'xyz'. so make sure we are not escaping very first quote at index 0.